wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/jpa/JpaWorkspaceDao.java

```diff
@@ -87,15 +87,17 @@ public WorkspaceImpl update(WorkspaceImpl update)
   }
 
   @Override
-  public void remove(String id) throws ServerException {
+  public Optional<WorkspaceImpl> remove(String id) throws ServerException {
     requireNonNull(id, "Required non-null id");
+    Optional<WorkspaceImpl> workspaceOpt;
     try {
-      Optional<WorkspaceImpl> workspaceOpt = doRemove(id);
+      workspaceOpt = doRemove(id);
       workspaceOpt.ifPresent(
           workspace -> eventService.publish(new WorkspaceRemovedEvent(workspace)));
     } catch (RuntimeException x) {
       throw new ServerException(x.getLocalizedMessage(), x);
     }
+    return workspaceOpt;
```

Comment:
Please add test in tck tests that should test made changes

Reply:
added